src/DiscoveryLocator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+declare(strict_types=1);
+
+namespace Http\Factory\Discovery;
+
+use InvalidArgumentException;
+use RuntimeException;
+
+abstract class DiscoveryLocator
+{
+    /** @var array  */
+    protected static $candidates = [];
+
+    /**
+     * @throws RuntimeException If no implementation is available
+     */
+    public static function locate(string $interface): string
+    {
+        foreach (static::$candidates[$interface] ?? [] as $candidate) {
```

Comment:
No need for `??` here, the type is declared as an array and has a default value.

Reply:
If you remove `?? []`, the test `testCannotLocateInvalidFactory` will fail. It is a public method, so we cannot assume that `$interface` actually is a key in the array.

Reply:
Ah, right. Still working on my first cup of tea. 😄